frb_codegen/src/generator/rust/mod.rs

```diff
@@ -26,13 +26,25 @@ use crate::others::*;
 
 pub const HANDLER_NAME: &str = "FLUTTER_RUST_BRIDGE_HANDLER";
 
+pub struct GeneratorOptions {
+    pub should_generate_sync_execution_mode_utility: Option<bool>
+}
+impl Default for GeneratorOptions {
+    fn default() -> Self {
+        Self { should_generate_sync_execution_mode_utility: Some(true) }
+    }
+}
+
 pub struct Output {
     pub code: String,
     pub extern_func_names: Vec<String>,
 }
 
-pub fn generate(ir_file: &IrFile, rust_wire_mod: &str) -> Output {
+pub fn generate(ir_file: &IrFile, rust_wire_mod: &str, options: Option<GeneratorOptions>) -> Output {
     let mut generator = Generator::new();
+    if options.is_some() {
+        generator.options = options.unwrap(); // can this be improved? 
```

Comment:
@fzyzcjy I'm still picking up rust, if there's a better way to do this, please suggest. Thanks

Reply:
What about sth like

```
let generator = Generator::new(options.unwrap_or_default());